src/middlewares/ExtractPDFMiddleware.js

```diff
@@ -0,0 +1,69 @@
+import fs from "fs";
+import path from "path";
+import PdfParse from "pdf-parse/lib/pdf-parse.js";
+import { Logger } from "../log/Logger.js";
+
+const __dirname = path.resolve();
+
+export class ExtractPDFMiddleware {
```

Comment:
Middleware의 일부 기능이 동일하다면, 추상화를 해서 이를 extends 를 받아서 사용하는 걸 추천

```suggestion
export class ExtractPDFMiddleware extends BaseMiddleware {
```

이렇게 하면 동일 작업에 대해서, 조금 더 직관적으로 구성하고, DuckTaping 을 진행해서 DI 와 같은 효과를 낼 수 있음.